ksql-engine/src/main/java/io/confluent/ksql/QueryEngine.java

```diff
@@ -58,11 +58,12 @@
 
   QueryEngine(
       final ServiceContext serviceContext,
+      final QueryIdGenerator queryIdGenerator,
```

Comment:
id generator is now injected to allow the sandbox to inject a id generator that will generate the same ids as a true `execute` call, i.e. this enables the `sandbox.validate` and `ksqlEngine.validate` calls to be inline when it comes to query ids